test/integration/unicorn/error-name-conflicts.js

```diff
@@ -0,0 +1,12 @@
+function foo() {
+	try {
+	} catch (err) {
+		console.log(err);
+
+		if (test) {
+			throw a;
+		} else {
+			throw b;
+		}
```

Comment:
We have a trouble here, the `catch-error-name` and `prevent-abbreviations` fix `err` to `error`, and this rule is creating `const error = test ? a : b;` cause `Parsing error: Identifier 'error' has already been declared`.

I have no idea how to fix this.

Reply:
Actual code  https://github.com/eslint/eslint/blob/b76aef778befb32afe7ad249934b132dc49713d2/lib/shared/config-validator.js#L128

Reply:
@futpib 

Glad to see you here, you are the one add `avoidCapture`(great util, BTW), do you have an idea how to fix this problem?

I was thinking use a global `Map` to manage the variables created by our rules, but I'm not sure if it will work. And this may still can't solve the real problem, if there is a new variable created by other rule from another plugin.

Reply:
Even if we make all rules in `eslint-plugin-unicorn` generate non-conflicting names, we can still hit name conflicts across eslint plugins, so I think this is a `eslint` issue.

Having a plugin-global map seems like an ok workaround. (It should probably be a `WeakMap<Scope, Set<string>>` like in `prevent-abbreviations` https://github.com/sindresorhus/eslint-plugin-unicorn/blob/16f6ef33c581782029bf33377f9084a0b2847fc1/rules/prevent-abbreviations.js#L552-L558))

Reply:
> we can still hit name conflicts across eslint plugins, so I think this is a eslint issue.

Can you open an issue on https://github.com/eslint/eslint ?

Reply:
https://github.com/eslint/eslint/issues/13706